### DISCUSSION TITLE
Maßstabsabhängige Darstellung bei Geoserver zieht nicht (Server Vendor aber richtig eingestellt)

### DISCUSSION THREAD

**Discussion by ue-black** [e-netze/webgis-community#90] [General]
Hallo!

Die maßstabsabhängige Sichtbarkeit unsererer WMS-Dienste scheint im Atlas noch nicht ganz zu funktionieren:

Wir haben im CMS folgendes eingestellt:
![cms](https://github.com/gis-eni/webgis-forum/assets/127403099/194193a4-36b2-4a84-b8d3-996e925f5b88)

Hier der Link zu einer Beispielkarte:
https://atlas.vorarlberg.at/portal/map/Transport%20und%20Verkehr/%C3%96ffentlicher%20Verkehr

Hier der WMS-Dienst der in der Karte eingebaut wurde. [ÖPNV-Steige](http://vogis.cnv.at/geoserver/vogis/wms?service=WMS&version=1.1.0&request=GetMap&layers=vogis%3Aoepnv_steige&bbox=-43201.80397%2C256130.81783%2C-42800.36231%2C256588.33010&width=674&height=768&srs=EPSG%3A31254&styles=&format=application/openlayers#toggle)

Im Geoserver ist eingestellt, dass die Sichtbarkeit bis 1:5000 geht:
![oepnv_gs](https://github.com/gis-eni/webgis-forum/assets/127403099/0f4ae212-e7eb-4af7-af53-c1a7283d2595)

Im Atlas sieht es aber so aus:
![oepnv_atlas](https://github.com/gis-eni/webgis-forum/assets/127403099/94260b8d-06ac-4f7e-a61a-3ed4f187e2eb)

Außerdem ist mir nicht ganz klar warum bei einem größeren Maßstab die Layer ausgegraut werden obwohl sie dargestellt werden:
![oepnv_atlas_grau](https://github.com/gis-eni/webgis-forum/assets/127403099/4a8df825-ff87-4634-8eff-f321db04956f)

Handelt es sich um einen Bug oder haben wir etwas falsch konfiguriert?

**Comment by gis-eni** [comment]
Der Dienst ist riesig und schwer zu handeln. Ich komme bei testen leider nicht weiter, weil ich immer wieder Fehler bekomme:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/2aae765e-b273-4f1f-a960-fbeacc256b1f)

Was mir auffällt ist erstens, der Dienst immer mit http aufgerufen wird. Da scheitet er Request teilweise. Auch wenn im CMS den Dienst mit https Einstelle wird im Capabilities XML immer wieder http verwiesen (OnlineResource für getmap usw).
Bitte das einmal richtig einstellen ;)

Zu Thema:
Wenn der Vendor ``GeoServer2_x`` eingestellt wird, schickt WebGIS den Paramater ``format_options=dpi:xxx`` mit, damit beide von der gleichen Auflösung ausgehen. Dann sollte die Maßstabsgrenzen in der Regel richtig sein. Allerdings kenn ich mich beim GeoServer nicht gut genug aus, um sagen zu können, dass das auch beim WMS 1.1.1 funktioniert. Verwendet bitte bei solchen Dingen immer erst einmal Version 1.3.0

**Comment by gis-eni** [comment]
Ok, nach weiteren Tests bin ich zu folgenden "Zwischen" Ergebnis gekommen:

Laut CMS bindest du den Dienst mit WMS Version 1.1.1 ein. Im Screenhot von oben zeigst du mir die Capabilities für den Layer mit Version 1.3.0. Dort gibt es ScaleMinDominator, ScaleMaxDominator...

In WMS 1.1.1 schaut das so aus für diesen Layer:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/9985090b-c954-4537-aa7c-24bdea0bf5e9)

Allerdings ist beim ScaleHint Max und Min per Definition nicht gleich dem ScaleDominator. Wie man den ScaleHint berechnet findet man zB hier:

https://wiki.deegree.org/deegreeWiki/HowToUseScaleHintAndScaleDenominator.html

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/2c643449-8bfe-4efb-b087-2957a49458c7)

Nachdem hier bei dir aber der gleiche Werte steht wie beim ScaleDominator kann hier schon was nicht passen.  Für 1:5000 müsste hier ca 2.0 raus kommen-

Binde den Dienst einmal mit Version 1.3.0 ein. Der ScaleHint sorgt anscheinend bei 1.1.1 zu verwirrung.

**Comment by gis-eni** [comment]
Bitte abklären, ob GeoServer den SkaleHint immer so interpretiert. Wenn ja, können wie das im WebGIS über den *Vendor* lösen. Damit sollte dann weiterhin WMS 1.11 funktionieren

**Reply by faulgurke** [reply]
Hier der Versuch von Antworten:
1. Wir haben an den Einstellungen für die GetCapabilities nichts geändert. Ist also keine speziell Vorarlberger GeoServer Variante.
2. Diese (fremden) WMS haben wir abgefragt. Überall kommt der ScaleHint vor (was nicht wirklich überrascht - ist ja Bestandteil der Spezifikation 1.1.1) - auch beim UMN MapServer kommt dieser Parameter vor.
https://gis.lfrz.gv.at/wmsgw/dopmaxresfree/?VERSION=1.1.1&REQUEST=GetCapabilities&SERVICE=WMS
https://vogis.cnv.at/mapserver/mapserv?map=i_landkarten_r_wms.map&version=1.1.1&service=wms&request=getcapabilities
3. Ob GeoServer den ScaleHint immer so interpretiert? Es erscheint uns zumindest plausibel, nachgerechnet haben wir nicht.

**Comment by gis-eni** [comment]
Gut, dann gehen wir einmal davon aus, dass euer GeoServer wie alle anderen reagiert.
Mit den neuem Build 5.23.3101 wird der Vendor, den du im CMS bei deinem Dienst angibst für den ScaleHint berücksichtig.
Bei GeoServer wird das dann einfach als ScaleDominator interpretiert. Der neue Build liegt bereits zum Download bereit
